pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -831,9 +832,16 @@ func UpdateLocationWithNodeLabels(location *[]string, nodeLabels map[string]stri
 	if len(invalidLabels) > 0 {
 		logger.Warningf("ignored invalid node topology labels: %v", invalidLabels)
 	}
-	for topologyType, value := range topology {
+
+	keys := make([]string, 0, len(topology))
```

Comment:
`make` the string with len 0 if you're going to append. Otherwise you'll end up with `len(topology)` empty strings which you'll end up appending to.

Reply:
Oops, nevermind. you were using `make([]string,0, capacity)`. My bad!